src/basic/rlimit-util.c

```diff
@@ -55,6 +61,39 @@ int setrlimit_closest(int resource, const struct rlimit *rlim) {
         return 0;
 }
 
+int setpriority_closest(int nice) {
+        int current, limit;
+        struct rlimit highest;
+
+        // Try to set requested nice level
+        if (setpriority(PRIO_PROCESS, 0, nice) == 0)
+                return 0;
+
+        // Permission failed
+        if (!IN_SET(errno, EPERM, EACCES))
+                return -errno;
+
+        errno = 0;
+        current = getpriority(PRIO_PROCESS, 0);
+        if (errno)
```

Comment:
nitpick: for numeric types we generally prefer comparison with "==", unless the type is actually "boolean-like", but errno really is not...

please use `if (errno != 0)…`

Reply:
done.